docs/source/checkpointing.rst

```diff
@@ -230,24 +230,138 @@ with `idx` parameter always unset, and the same :class:`~.Function` can only be
 loaded using the same mode.
 
 
-Using disk checkpointing in adjoint simulations
+Using checkpointing in adjoint simulations
 ===============================================
 
 When adjoint annotation is active, the result of every Firedrake operation is
-stored in memory. For some simulations, this can result in a very large memory
-footprint. As an alternative, it is possible to specify that those intermediate
-results in forward evaluations of the tape which have type
-:class:`~firedrake.function.Function` be written to disk. This is usually the
-bulk of the data stored on the tape so this largely alleviates the memory
-problem, at the cost of the time taken to read to and write from disk.
+stored in memory. For some time-dependent simulations, this can lead to a
+large memory footprint. To alleviate this, checkpointing strategies can be used
+to store intermediate results either in memory or on disk.
```

Comment:
This reads a bit odd. It seems to be saying "to stop excessive memory usage we can store intermediate results in memory". Probably better to say something like "we can use checkpointing strategies to store only some intermediate results in memory or on disk".